src/table/components/head/ColumnAdjuster.tsx

```diff
@@ -21,48 +23,84 @@ const ColumnAdjuster = ({ column, isLastColumn }: AdjusterProps) => {
 
   if (!applyColumnWidths || constraints.active) return null;
 
-  const mouseMoveHandler = (evt: MouseEvent) => {
-    const deltaWidth = evt.clientX - tempWidths.current.initX;
-    const adjustedWidth = Math.max(tempWidths.current.initWidth + deltaWidth, MIN_COLUMN_WIDTH);
-
+  const updateWidth = (adjustedWidth: number) => {
     tempWidths.current.columnWidth = adjustedWidth;
     const newColumnWidths = [...columnWidths];
     newColumnWidths[pageColIdx] = adjustedWidth;
     setColumnWidths(newColumnWidths);
   };
 
-  const mouseUpHandler = (evt: MouseEvent) => {
-    evt.preventDefault();
-    evt.stopPropagation();
-    document.removeEventListener('mousemove', mouseMoveHandler);
-    document.removeEventListener('mouseup', mouseUpHandler);
-
+  const confirmWidth = () => {
     if (tempWidths.current.columnWidth !== tempWidths.current.initWidth) {
       const newWidthData = { type: ColumnWidthTypes.PIXELS, pixels: tempWidths.current.columnWidth };
       applyColumnWidths(newWidthData, column);
     }
   };
 
+  const mouseMoveHandler = (evt: MouseEvent) => {
+    const deltaWidth = evt.clientX - tempWidths.current.initX;
+    const adjustedWidth = Math.max(tempWidths.current.initWidth + deltaWidth, MIN_COLUMN_WIDTH);
+    updateWidth(adjustedWidth);
+  };
+
+  const mouseUpHandler = (evt: MouseEvent, adjuster: HTMLDivElement) => {
```

Comment:
do we really need to pass the adjuster? Can you use the evt.target or is that sometimes the wrong element?

Reply:
that sometimes is the wrong element

Reply:
ok then could we store it on the `tempWidths` ref, so you don't have to pass it to the mouseup handler and risk a memory leak

Reply:
Sure. Done